### ISSUE TITLE
Material Date-Picker

### ISSUE THREAD

**Issue by subratawebops** [formly-js/ngx-formly#614]
I don't see an option for date-picker. Your documentation have got no example for date-picker.

**Comment by aitboudad** [comment]
well you can create your own type, right now we only support the basic type but this will not prevent us to add more types.
Right I suggest you to create your own type as this will take some time to be implemented anyway I'll provide you an example!

**Comment by subratawebops** [comment]
@aitboudad It will be great if you can provide an example. I'll create mine. Thank you

**Comment by aitboudad** [comment]
see https://stackblitz.com/edit/angular-gkwj6z

**Comment by subratawebops** [comment]
@aitboudad Thank you very much

**Comment by ho0** [comment]
@aitboudad On a page where I have multiple forms (tasks) and a due date for each form (task), the date picker toggle doesn't work right. It actually attaches to the last date field. When I use the date picker to set the date, it only ever sets the last date on the page. The date field works if I don't use the date picker control.

**Comment by jogelin** [comment]
@aitboudad Do you planify to implement it ?

Do you have an idea how to you manage the validation of the date when there is a wrong value in the field but the field is not empty ? 

Because the datepicker of angular material return null when the date is not valid but it doesn't mean that the input is empty....

I would like to display an error message when the field is empty and when the field contain an invalid date but it seems it is not possible because the material datepicker always return a null value if the date is not valid....

My example : [https://stackblitz.com/edit/angular-datepicker-custom-adapter?file=app/date-adapter.component.ts](https://stackblitz.com/edit/angular-datepicker-custom-adapter?file=app/date-adapter.component.ts)
You can specify a wrong format and you will always see the required error + the matDatepickerParse

**Comment by aitboudad** [comment]
@jogelin don't use required and instead define a custom validator for date type.

> Do you planify to implement it ?

yeah but with a low priority now.

**Comment by aitboudad** [comment]
done, will be landed in `v3.0`:

```ts
import { FormlyMatDatepickerModule } from '@ngx-formly/material/datepicker';

@NgModule({
  imports: [
    ...
    FormlyMaterialModule,
    FormlyMatDatepickerModule
  ],
})
export class AppModule { }
```
